client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamImpl.java

```diff
@@ -187,7 +187,9 @@ private void failConnection(Throwable throwable) {
                     log.warn("Connection for segment {} failed due to: {}", segmentName, throwable.getMessage());
                 }
             }
-            if (failSetupConnection) {
+            if (throwable instanceof SegmentSealedException || throwable instanceof NoSuchSegmentException) {
+                setupConnection.releaseExceptionally(throwable);
```

Comment:
Is this throwable going to be exposed via the client API? I'd rather not expose `SegmentSealed` and `NoSuchSegment` as the application wouldn't really know how to react to them.

Reply:
What should it do? The writes can never succeed. Before it was simply doing nothing other than logging them leaving the futures uncompleted. That is the bug this PR is fixing. We obviously can't have them succeed. So we need to throw something that helps debug the problem. It's obviously runtime as we don't expect anyone to handle the exception. Additionally I don't want to name the exception after a speculative cause, as that may not be the true reason for the error.

Reply:
When I present Pravega, I actually emphasize a lot segment and sealing, so these are concepts that shouldn't be foreign to a Pravega developer. Let's leave it for now.